### PR TITLE
integrate axios to start using with the application for new api requests

### PR DESCRIPTION
**Pull request by ricardofreitasrocha**
### JIRA link (if applicable) ###
N/A

### Change description ###
Add axios package and request helpers to start using for new api requests.
Add a sample usage for `fetchCourts` and `fetchAllCourts`

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
[x] Maybe
```

### CODE REVIEW

**Comment by gregs-cgi** [comment]
I probably shouldn't review this as I did the work on gitlab :D
I would suggest we should get as much of the work updating controllers across as we can alongside these changes.

**Comment by ricardofreitasrocha** [comment]
> I probably shouldn't review this as I did the work on gitlab :D I would suggest we should get as much of the work updating controllers across as we can alongside these changes.

Fair, I can get someone else to review.

About the other changes, they're being done as part of https://github.com/hmcts/juror-bureau/tree/feature/JM-5346 ... I still think that we should have this initial implementation merged and any new requests to the api being done through it, whilst refactoring the existing dao objects.

File: server/objects/dataAccessObject.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  // type Method = 'delete' | 'get' | 'patch' | 'post' | 'put';
+  // type AxiosClientArgs = {
+  //   uri?: string,
+  //   body?: any,
+  //   headers?: {[key:string]: string},
+  //   transform?: (any) => any;
+  //   debug?: Promise;  // If provided, this promise is always returned instead of calling the API
+  // }
+  // type ICustomFunctions = Partial<{[key in Method]: (...args) => AxiosClientArgs;}>
```
Comment (callum-r-young) [inline server/objects/dataAccessObject.js:14]: Any reason this is commented out?

Comment (ricardofreitasrocha) [reply server/objects/dataAccessObject.js:14]: yes, these are types 😃

**Review by callum-r-young — APPROVED** [review]

**Pull request merged**